src/styles/_catalog.scss

```diff
@@ -1,22 +1,30 @@
+$mainSize: 200px;
+
 .catalog {
-  display: flex;
-  flex-wrap: wrap;
+  display: grid;
+  grid-template-columns: $mainSize;
+  justify-content: center;
+  gap: 48px;
+
+  @media (min-width: 488px) {
+    grid-template-columns: $mainSize $mainSize;
+  }
+
+  @media (min-width: 768px) {
+    grid-template-columns: $mainSize $mainSize $mainSize;
+  }
 
-  max-width: 944px;
-  margin: 0 auto;
+  @media (min-width: 1024px) {
+    grid-template-columns: $mainSize $mainSize $mainSize $mainSize;
```

Comment:
You can create a variable with the count of repeat columns and use a loop for media, but it's optional.